juju/juju
# Use mongo 4.4 on k8s

When bootstrapping on k8s, use the mongo 4.4 image.
Also replace deprecated SSL parameters with TLS ones.

## QA steps

juju bootstrap microk8s
juju bootstrap lxd

In both cases, double check that mongo 44 is the version of mongo that is running.

For the LXD controller
juju create-backup

juju bootstrap a 2.8 controller
juju upgrade-controller --build-agent

Check that the upgrade completes
ssh to the controller and cat /etc/systemd/system/juju-db.service
check the service args are the legacy ones
ExecStart=/usr/bin/mongod --auth --bind_ip_all --dbpath '/var/lib/juju/db' --ipv6 --journal --keyFile '/var/lib/juju/shared-secret' --oplogSize 1024 --port 37017 --quiet --replSet juju --sslMode requireSSL --sslPEMKeyFile '/var/lib/juju/server.pem' --sslPEMKeyPassword=ignored --storageEngine wiredTiger --syslog
